### ISSUE TITLE
Running trinucleotideMatrix generates Error in.Call2

### ISSUE THREAD

**Issue by Olsennnnn** [PoisonAlien/maftools#1015]
Hi,
While running, I got a mistake, like`

```r
laml.tnm = trinucleotideMatrix(maf = maf,ref_genome = "BSgenome.Hsapiens.UCSC.hg38")
-Extracting 5' and 3' adjacent bases
Error in .Call2("C_solve_user_SEW", refwidths, start, end, width, translate.negative.coord,  : 
  solving row 32714: 'allow.nonnarrowing' is FALSE and the supplied start (242366418) is > refwidth + 1`.
```

**sessionInfo**

```r
R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Asia/Shanghai
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome.Hsapiens.UCSC.hg38_1.4.5
 [3] BSgenome_1.70.1                   rtracklayer_1.62.0               
 [5] BiocIO_1.12.0                     Biostrings_2.70.1                
 [7] XVector_0.42.0                    GenomicRanges_1.54.1             
 [9] GenomeInfoDb_1.38.5               IRanges_2.36.0                   
[11] S4Vectors_0.40.2                  BiocGenerics_0.48.1              
[13] maftools_2.16.0                  

loaded via a namespace (and not attached):
 [1] Matrix_1.6-4                compiler_4.3.1              rjson_0.2.21               
 [4] crayon_1.5.2                SummarizedExperiment_1.32.0 Biobase_2.62.0             
 [7] Rsamtools_2.18.0            bitops_1.0-7                GenomicAlignments_1.38.0   
[10] parallel_4.3.1              splines_4.3.1               BiocParallel_1.36.0        
[13] yaml_2.3.8                  lattice_0.22-5              DNAcopy_1.76.0             
[16] S4Arrays_1.2.0              XML_3.99-0.16               DelayedArray_0.28.0        
[19] MatrixGenerics_1.14.0       GenomeInfoDbData_1.2.11     RColorBrewer_1.1-3         
[22] SparseArray_1.2.3           zlibbioc_1.48.0             grid_4.3.1                 
[25] rstudioapi_0.15.0           data.table_1.15.2           codetools_0.2-19           
[28] survival_3.5-7              abind_1.4-5                 RCurl_1.98-1.13            
[31] restfulr_0.0.15             matrixStats_1.2.0           tools_4.3.1
```

Because I am using `Homo_sapiens. GRCh38. DNA. Primary_assembly` annotations, so I think should use GRCh38 in trinucleotideMatrix function.

Do you have any suggestion how to solve it?
Thanks
Olsen

**Comment by Olsennnnn** [comment]
Could anyone help me?Thanks.

**Comment by PoisonAlien** [comment]
Hi Olsen,

Thank you for the issue and, sorry for the tardy response. The issue seems to be due to the variant position being larger than the chromosomal length.  Could you please check what chromosome the variant loci `242366418` is located on?

**Comment by Olsennnnn** [comment]
Hi,
Thank you for your answer.
I looked at my maf file and found that there was no variation at loci`242366418` in the file, but the strange thing was that there was variation at loci`242366419` adjacent to loci`242366418`, and loci`242366419` was located on chromosome 2.
However, when I run the `trinucleotideMatrix` function again, the same error as above still occurs, indicating the loci`242366418`.

Looking forward to your reply.
Thanks
Olsen

**Comment by ShixiangWang** [comment]
@PoisonAlien I think this is a mismatch error between the reference genome and input data. I recorded the same error last year (with reproduciable example https://github.com/ShixiangWang/sigminer/issues/422) but did not handle it as it hardly happened.

**Comment by PoisonAlien** [comment]
Thank you @ShixiangWang for the response. 

I think there is a mismatch here. `chr2` length from BSgenome.hg38 is smaller than the variant of interest here. 

```
> seqlengths(BSgenome.Hsapiens.UCSC.hg38)["chr2"]
     chr2 
242193529 

> seqlengths(BSgenome.Hsapiens.UCSC.hg38)["chr2"] < 242366418
chr2 
TRUE 
```

@Olsennnnn Please make sure you are using the correct reference genome. You can also check this with the fasta file that was used for the variant calling. Look for `chr2` length in the `fai` file and make sure the lengths are appropriate.

**Comment by Olsennnnn** [comment]
Thank you @PoisonAlien @ShixiangWang for the response.
My process goes like this (hopefully this provides you with useful information) : 

First of all, I converted a `vcf` file to a `maf` file using the `vcf2maf` function.In this process, I used the `ref - fasta` : `Homo_sapiens. GRCh38. DNA. Primary_assembly. Fa`.So does this mean that the error based on the above, I should change `Homo_sapiens. GRCh38. DNA. Primary_assembly. Fa` to `Homo_sapiens. GRCh37. DNA. Primary_assembly. Fa` in this step?

Secondly,through your code, I checked respectively `chr2` length in `BSgenome Hsapiens. UCSC. Hg38` and `BSgenome Hsapiens. UCSC. Hg19`, I found loci`242366418` is greater than the length of the `chr2` in `BSgenome Hsapiens. UCSC. Hg38`. But in `BSgenome Hsapiens. UCSC. Hg19` ,loci`242366418` is smaller than the length of `chr2`, does that mean I should use `BSgenome Hsapiens. UCSC. Hg19` instead of `BSgenome Hsapiens. UCSC. Hg38`?

So I should make changes in these two places(1.When running `vcf2maf`, change `Homo_sapiens. GRCh38. DNA. Primary_assembly. Fa` to `Homo_sapiens. GRCh37. DNA. Primary_assembly. Fa` in `ref - fasta`.2.When running `trinucleotideMatrix`, change `BSgenome Hsapiens. UCSC. Hg38` to `BSgenome Hsapiens. UCSC. Hg19` in `ref_genome`.), am I getting this right?

**Comment by ShixiangWang** [comment]
@Olsennnnn I think there is another info you should provide: which genome build did you use for aligning your fastq data and generating the VCF records?

**Comment by Olsennnnn** [comment]
Hi,
Sorry, I don't know which genome build I used.This file was created a long time ago, so my colleagues they forgot too.
In addition, I generated a new maf file with `GRCh37`.The new maf file still generates an error when running `trinucleotideMatrix`, like this:
```
-Extracting 5' and 3' adjacent bases
Error in .Call2("C_solve_user_SEW", refwidths, start, end, width, translate.negative.coord,  : 
  solving row 46154: 'allow.nonnarrowing' is FALSE and the supplied start (198083718) is > refwidth + 1
```
loci`198083718` was located on chromosome 3.
When I examined the maf file, I found more than one variant loci beyond chromosome length.
Why is that? How can I fix this?

Thanks
Olsen
